ext/json/json.stub.php

```diff
@@ -2,7 +2,7 @@
 
 /** @generate-class-entries */
 
-function json_encode(mixed $value, int $flags = 0, int $depth = 512): string|false {}
+function json_encode(mixed $value, int $flags = 0, int $depth = 512, string|int $indent = 4): string|false {}
```

Comment:
I think accepting string only here is enough.

Reply:
It does make it more difficult to see that `$indent='    '` specifies 4 spaces and that this indentation is actually consistent with some other call site that uses `$indent=4`.

Also, I prefer parameters having pure types rather than composite. Since we have named parameters now it might make sense to split into two parameters:

`json_encode($value, indent: 4, indent_char: ' ');`

Reply:
I agree with @dtakken about pure types rather than composite. As I was scanning the code I kept trying to figure out why the `string|int` types were being used instead of only `int`. Realizing finally if a `string` is passed as the argument the `string` **literally** is the indent value.
However, spiting this into two parameters does _remove_ or _complicate_ the feature of multiple characters:
```php 
json_encode($value, indent: 4, indent_chars: '🚀🌙🌠🌎');
// What to do if indent = 2 and indent_chars has more than 2 chars?
json_encode($value, indent: 2, indent_chars: '🚀🌙🌠'); // error? ignore the 3rd char?
// What to do if indent = 3 and indent_chars has less than 3 chars?
json_encode($value, indent: 3, indent_chars: '🚀🌙'); // repeat '🚀🌙🚀'? error? something else?
```